x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -38,6 +40,18 @@ export async function createDatasource(options: {
   await baseSetup(callCluster);
   const pkg = await Registry.fetchInfo(pkgkey);
 
+  // Collect the config for each dataset
+  const streams: Stream[] = [];
```

Comment:
How about moving this to a function and doing something like 

```typescript
const streams = await getStreams(pkg)
```

Reply:
replacing comments by self descriptive code 👍